src/NFCManager.cpp

```
@@ -1630,6 +1632,139 @@ bool NFCManager::executeWrite(const NFCWriteRequest& request) {
+        // Single-pass NFC write — no sequential drops, no scan loop race
+        Serial.println("NFCManager: WRITE_ATOMIC writing to NFC...");
+        opt_nfc_hal_t* hal = connection_->getHal();
+        err = opt_write_to_nfc(&writeScratchTag_, hal);
+        if (err != OPT_OK) {
+            Serial.printf("NFCManager: WRITE_ATOMIC NFC write failed: %s\n", opt_error_str(err));
+            return false;
+        }
+
+        // Verify with retries (same pattern as formatNewSpool)
+        const int maxRetries = 3;
+        for (int retry = 0; retry < maxRetries; retry++) {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)